lib/concurrent/synchronization/java_pure_object.rb

```diff
@@ -32,6 +33,10 @@ def ns_signal
           JRuby.reference0(self).notify
           self
         end
+
+        def ensure_ivar_visibility!
+          # relying on undocumented behavior of JRuby, ivar access is volatile
```

Comment:
> But whether or not that's true - how about we simply make it documented behaviour by agreeing it with the JRuby team and getting it written down somewhere?

Are you talking about making all Ruby i-vars volatile? Isn't it going to have terrible performance consequences?